site/web/assets/js/classification.js

```diff
@@ -21,24 +22,21 @@ var insertClassification = function (classificationDetails) {
       var value = $('#category-' + count).val();
       if (value !== 'NULL') {
         var increase = false;
-        var id = 0;
-        for (var i = 0; i < classificationDetails.length; i++) {
-          if (classificationDetails[i]['name'] === value) {
-            id = classificationDetails[i]['id'];
-            break;
-          }
-        }
+        var selected = $(this).find('option:selected');
+        var id = selected.data('id');
         for (var i = 0; i < classificationDetails.length; i++) {
           if (classificationDetails[i]['parentId'] === id) {
             if (increase === false) {
               var category = $('#category-' + count).clone().appendTo('.classification-system');
               count++;
               category.attr('id', 'category-' + count).empty().append('<option value="NULL">select a category</option>');
-              $('#category-' + count).append($('<option>').val(classificationDetails[i]['name']).html(classificationDetails[i]['name']));
+              $('#category-' + count).append($('<option>').val(classificationDetails[i]['name']).html(classificationDetails[i]['name'])
+                .attr('data-id',classificationDetails[i]['id']));
```

Comment:
Please reduce the length of these lines to a sane ~ 80-100 characters, it's not even fitting the GitHub diff view, which makes review annoying.
And add a space after the ",", e.g. `.attr('data-id', classificationDetails[i]['id'])`

(This also applies to other places in this file, please fix accordingly.)